src/main/scala/castalia/Receptionist.scala

```diff
@@ -17,8 +17,9 @@ class Receptionist extends Actor with ActorLogging {
 
   val metricsCollector = createMetricsCollector
 
-  private def upsertEndPointActor(stubConfig: StubConfig, endpointMatcher : RequestMatcher) = {
+  val requestMatcherActor = context.actorOf(RequestMatcherActor.props())
 
+  private def upsertEndPointActor(stubConfig: StubConfig) = {
     def endpointActorFactory(stubConfig: StubConfig): JsonEndpointActor = {
       if (stubConfig.responseprovider.isDefined) new JsonResponseProviderEndpointActor(stubConfig, metricsCollector)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/scala-academy/Castalia/file/1993177510/issues/source?bid=2998213&fileBranchId=3120054#l24)